include/onnxruntime/core/providers/openvino/openvino_provider_factory.h

```diff
@@ -5,13 +5,23 @@
 
 #ifdef __cplusplus
 extern "C" {
+#else
+#include <stdbool.h>
 #endif
 
 /**
- * \param device_id openvino device id, starts from zero.
+ * \param device_type openvino device type and precision. Could be any of
+ * CPU_FP32, GPU_FP32, GPU_FP16, MYRIAD_FP16, VAD-M_FP16 or VAD-F_FP32.
  */
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_OpenVINO,
-    _In_ OrtSessionOptions* options, const char* device_id);
+    _In_ OrtSessionOptions* options, _In_ const char* device_type);
+
+/**
+ * \param settings_str string of Key-Value pairs with '\n' used to delimit
+ * pairs and '|' used to delimit key and value within a pair.
+ */
```

Comment:
Also, if it works for openvino, you can use the same format as Nuphar and reuse the tokenization (parsing) code.

Reply:
Nuphar setting strings use ':' and ',' as delimiters, which, in OpenVINO EP, are also used as config values. This makes Nuphar's settings format inapplicable to OpenVINO.